### PR TITLE
Update failing RPC example

### PR DESCRIPTION
**Pull request by lorisleiva**
This PR fixes a failing example due to a recent refactoring that uses an `execute` function as the RPC Plan returned by the RPC API.

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3442?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3442** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3442?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3435** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3435?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lorisleiva and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by mcintyre94 — APPROVED** [review]
Nice! It's good to have an example of how this looks with a custom API too

File: examples/rpc-custom-api/src/example.ts [hunk 1]
```diff
@@ -76,8 +78,8 @@ const customizedRpcApi = new Proxy(solanaRpcApi, {
             /**
```
Comment (steveluscher) [inline examples/rpc-custom-api/src/example.ts:78]: <i class='graphite__hidden'>[Re: line +75]<br/><br/></i>

```ts
    get(target, p, receiver): (address: Address) => RpcPlan<AssetMetadata> {
```<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3442?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

File: examples/rpc-custom-api/src/example.ts [hunk 2]
```diff
@@ -76,8 +78,8 @@ const customizedRpcApi = new Proxy(solanaRpcApi, {
...
              * When the `getAssetMetadata` method is called on the RPC, return a custom definition.
              */
-            return (address: Address) => {
-                return {
+            return (address: Address): RpcPlan<AssetMetadata> => {
```
Comment (steveluscher) [inline examples/rpc-custom-api/src/example.ts:81]:
```suggestion
            return (address) => {
```

**Review by steveluscher — APPROVED** [review]

**Pull request merged**